app/get_proc_arg.go

```diff
@@ -0,0 +1,237 @@
+package app
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+
+	"github.com/rancher/longhorn-manager/types"
+)
+
+const (
+	ArgFlexvolumePluginDir = "volume-plugin-dir"
+	ArgRootDir             = "root-dir"
+
+	DefaultRootDir       = "/var/lib/kubelet"
+	DefaultFlexvolumeDir = "/usr/libexec/kubernetes/kubelet-plugins/volume/exec/"
+
+	DetectPodMaxPolls     = 120
+	DetectFlexvolumeName  = "discover-flexvolume-dir"
+	DetectRootDirName     = "discover-kubelet-root-dir"
+	FindKubeletProcScript = `
+    find_kubelet_proc() {
+      for proc in $(find /proc -maxdepth 1 -type d 2>/dev/null); do
+        if [ ! -f $proc/cmdline ]; then
+          continue
+        fi
+        if [[ "$(cat $proc/cmdline | tr '\000' '\n' | head -n1 | tr '/' '\n' | tail -n1)" == "kubelet" ]]; then
+          echo $proc
+        fi
+      done
+    }
+  	`
+	DetectFlexVolumeScript = FindKubeletProcScript + `
```

Comment:
the only difference between `get_flexvolume_path()` and `get_kubelet_root_dir()` is what to grep and what's the default. The key for grepping can be set as a parameter, and what's the default can be covered by caller when returned path is "".

get `data-dir` and `-d`is similar.